### PR TITLE
Precompile events - Native Minter

### PR DESCRIPTION
**Pull request by abi87**
## Why this should be merged

Adds event to mint function. 

## How this works

Adds new Event to ABI and regenerated with the precompile generation tool.

`event NativeCoinMinted(address indexed sender, address indexed recipient, uint256 amount);`

Increases total gas cost by 1756 (base event gas, 3 topics gas, 32-byte unindexed data)

## How this was tested

Added UT and hardhat tests.

## How is this documented

Will be documented with other Durango changes

### CODE REVIEW

File: contracts/contracts/interfaces/INativeMinter.sol
Comment (richardpringle) [inline contracts/contracts/interfaces/INativeMinter.sol:1]: 🚀

File: precompile/contracts/nativeminter/contract_test.go [hunk 1]
```diff
@@ -158,6 +160,58 @@ var tests = map[string]testutils.PrecompileTest{
...
+		SuppliedGas: MintGasCost,
+		ReadOnly:    false,
+		ExpectedRes: []byte{},
+		AfterHook: func(t testing.TB, stateDB contract.StateDB) {
+			// Check no logs are stored in state
+			logsTopics, logsData := stateDB.GetLogData()
+			require.Len(t, logsTopics, 0)
+			require.Len(t, logsData, 0)
+		},
+	},
```
Comment (aaronbuchwald) [inline precompile/contracts/nativeminter/contract_test.go:185]: Isn't this the same as "mint funds from enabled address" above?

**Review by aaronbuchwald — CHANGES_REQUESTED** [review]

File: contracts/test/contract_native_minter.ts [hunk 2]
```diff
@@ -31,8 +36,26 @@ describe("ERC20NativeMinter", function () {
...
+
+describe("INativeMinter", function () {
+  let owner: SignerWithAddress
+  let contract: Contract
+  before(async function () {
+    owner = await ethers.getSigner(ADMIN_ADDRESS);
+    contract = await ethers.getContractAt("INativeMinter", MINT_PRECOMPILE_ADDRESS, owner)
+  });
+
+  it("should emit reward address changed event ", async function () {
```
[the remark this answers is not in the document]
Comment (ceyonur) [reply contracts/test/contract_native_minter.ts:54]: copy 🍝

File: precompile/contracts/nativeminter/contract_test.go (hunk 1 above)
Comment (ceyonur) [reply precompile/contracts/nativeminter/contract_test.go:185]: indeed, they're just testing different after hooks and have different focus. I'd like to keep them as they're for the clarity.

File: contracts/test/contract_native_minter.ts [hunk 3]
```diff
@@ -3,6 +3,11 @@
 
 import { ethers } from "hardhat"
 import { test } from "./utils"
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import {
+  Contract,
+} from "ethers"
```
Comment (darioush) [inline contracts/test/contract_native_minter.ts:10]:
```suggestion
import { Contract } from "ethers"
```

**Review by darioush — APPROVED** [review]

File: precompile/contracts/nativeminter/event.go [hunk 4]
```diff
@@ -12,11 +12,23 @@ import (
 )
 
 const (
-	NativeCoinMintedEventGasCost = contract.LogGas + contract.LogTopicGas*3
+	// NativeCoinMintedEventGasCost is the gas cost of the NativeCoinMinted event.
+	// It is the base gas cost + the gas cost of the topics (signature, sender, recipient)
+	// and the gas cost of the non-indexed data (32 bytes for amount).
+	NativeCoinMintedEventGasCost = contract.LogGas + contract.LogTopicGas*3 + contract.LogDataGas*32
```
Comment (aaronbuchwald) [inline precompile/contracts/nativeminter/event.go:18]:
```suggestion
	NativeCoinMintedEventGasCost = contract.LogGas + contract.LogTopicGas*3 + contract.LogDataGas*common.HashLength
```

**Review by aaronbuchwald — DISMISSED** [review]

**Review by aaronbuchwald — APPROVED** [review]

**Pull request merged**